common/clap_utils/src/lib.rs

```diff
@@ -157,3 +165,65 @@ pub fn parse_ssz_optional<T: Decode>(
         })
         .transpose()
 }
+
+/// Attempts to parse a file into a `HashMap<String,String>`. Requires the filename end in `.yaml`/`.yml`
+/// for a YAML config file or `.toml` for a TOML config file.
+pub fn parse_file_config(file_name: &str) -> Result<HashMap<String, String>, String> {
+    if file_name.ends_with(".yaml") || file_name.ends_with(".yml") {
+        fs::read_to_string(file_name)
+            .map_err(|e| e.to_string())
+            .and_then(|yaml| serde_yaml::from_str(yaml.as_str()).map_err(|e| e.to_string()))
+            .and_then(|parsed| to_string_map(parsed, yaml_value_to_string))
+    } else if file_name.ends_with(".toml") {
+        fs::read_to_string(file_name)
+            .map_err(|e| e.to_string())
+            .and_then(|toml| toml::from_str(toml.as_str()).map_err(|e| e.to_string()))
+            .and_then(|parsed| to_string_map(parsed, toml_value_to_string))
+    } else {
+        Err("config file must have extension `.yml`, `.yaml` or `.toml`".to_string())
+    }
+}
+
+fn to_string_map<V, F: Fn(V) -> Result<String, String>>(
+    map: HashMap<String, V>,
+    f: F,
+) -> Result<HashMap<String, String>, String> {
+    let mut new_map = HashMap::new();
+    for (key, value) in map.into_iter() {
+        new_map.insert(key, f(value)?);
+    }
+    Ok(new_map)
+}
+
+fn toml_value_to_string(value: TomlValue) -> Result<String, String> {
```

Comment:
just trying to understand, why not just use a Serializer?

Reply:
So I first tried out parsing the whole config file directly to a `HashMap<String, String>` but in that case, the TOML config file had to be entirely strings and YAML arrays had to be strings which is why I went the route of parsing to the TOML/YAML `Value` structs. 

I tried using `toml::to_string(&v)` rather this this custom `toml_value_to_string` method but it formats things pretty poorly for example: 

```
#[test]
fn test() {
    assert_eq!(
        toml::to_string(&TomlValue::String("debug".to_string())).unwrap(),
        "\"debug\""
    );
}
```